servers/src/main/java/org/jboss/aerogear/unifiedpush/auth/PathBasedKeycloakConfigResolver.java

```diff
@@ -48,49 +52,55 @@
 
 	@Override
 	public KeycloakDeployment resolve(Request request) {
-		String realm = DEFAULT_REALM_FILE;
+		String fileName = DEFAULT_REALM_FILE;
+		String jwtRealm = OAuth2Configuration.DEFAULT_OAUTH2_UPS_REALM;
 
 		AccessToken accessToken = BearerHelper.getTokenDataFromBearer(request).orNull();
 		if (accessToken != null) {
 			String issuer = accessToken.getIssuer();
 
 			if (StringUtils.isNoneEmpty(issuer)) {
 				issuer = URLUtils.removeLastSlash(issuer);
-				String jwtRealm = URLUtils.getLastPart(issuer);
+				jwtRealm = URLUtils.getLastPart(issuer);
 
 				if (!OAuth2Configuration.DEFAULT_OAUTH2_UPS_REALM.equalsIgnoreCase(jwtRealm)) {
-					realm = "upsi";
+					fileName = "upsi";
 				}
 			}
 		}
 
 		URI uri = URI.create(request.getURI());
 
 		if (logger.isTraceEnabled())
-			logger.trace("Identified Bearer request, using keycloak realm! URI: {}, realm: {}", uri.toString(), realm);
+			logger.trace("Identified Bearer request, using keycloak realm! URI: {}, realm: {}", uri.toString(), jwtRealm);
 
 		String host = uri.getHost();
-		CustomKeycloakDeployment deployment = cache.get(getCacheKey(realm, host));
+		CustomKeycloakDeployment deployment = cache.get(getCacheKey(jwtRealm, host));
 		if (null == deployment) {
 			InputStream is = null;
 
 			try {
-				is = applicationContext.getResource("/WEB-INF/" + realm + ".json").getInputStream();
+				is = applicationContext.getResource("/WEB-INF/" + fileName + ".json").getInputStream();
 			} catch (IOException e) {
-				throw new IllegalStateException("Not able to find the file /" + realm + ".json");
+				throw new IllegalStateException("Not able to find the file /" + fileName + ".json");
 			}
 
 			if (is == null) {
-				throw new IllegalStateException("Not able to find the file /" + realm + ".json");
+				throw new IllegalStateException("Not able to find the file /" + fileName + ".json");
```

Comment:
please make the two messages slightly different so they'd be able to be differentiated in the log